Assets/Engine/Interactions/DelayedInteraction.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Mirror;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace SS3D.Engine.Interactions
+{
+    /// <summary>
+    /// Base class for interactions which occur after a delay
+    /// </summary>
+    public abstract class DelayedInteraction : /* Hack (no multiple inheritance) */ NetworkBehaviour, ContinuousInteraction
+    {
+        /// <summary>
+        /// The delay in seconds before the interaction is performed
+        /// </summary>
+        public float Delay;
+        private float startTime;
+
+        /// <summary>
+        /// The prefab to use as a loading bar. If this value is null no bar is shown.
+        /// </summary>
+        public GameObject LoadingBarPrefab;
+        
+        public abstract InteractionEvent Event { get; set; }
+        public abstract string Name { get; }
+
+        public virtual bool CanInteract()
+        {
+            // Prevents multiple people from interacting at the same time
+            // Is this a good idea?
+            return startTime == 0;
```

Comment:
startTime is not synced between different instances of DelayedInteraction, so this will always be `true`.
If you really want to ensure that no other person can interact with an object at the same time as you, add an empty component to the gameobject when you start to interact (e.g. a `new CurrentlyInteracting(typeof(DelayedInteraction))`) and in `CanInteract` check whether or not that component is present on the object

Reply:
This just prevents multiple interactions on the same object (ex. wrench). It is not meant to block any other kind of interaction.
The whole thing is just here to prevent weird state changes while waiting for the delay. It should be replaced with some better system down the line.

Reply:
At best this will prevent same interactions from the same client though, and even then that's not supported behaviour. If it's being replaced then sure that's fine, but yeah keep in mind this won't even be replicated on the server and will just be for this client only.

Reply:
No it will not, interactions run on the server?
Your comment made me realize that the client will still see the option, but not be able to perform it. In the end, this will need to be replaced, because it's hacky and I'm not sure how multiple interactions are supposed to work.